content/blog/ellmer-0-4-0/index.Rmd

```diff
@@ -0,0 +1,102 @@
+---
+output: hugodown::hugo_document
+
+slug: ellmer-0-4-0
+title: ellmer 0.4.0
+date: 2025-11-13
+author: Hadley Wickham
+description: >
+    ellmer 0.4.0 includes important lifecycle updates, new Claude features
+    (caching, file uploads, web tools), OpenAI improvements, and enhancements
+    to error handling, pricing tracking, and security.
+photo:
+  url: https://unsplash.com/photos/a-herd-of-elephants-standing-next-to-each-other-CzIwSXedUGM
+  author: Evan Jones
+
+# one of: "deep-dive", "learn", "package", "programming", "roundup", or "other"
+categories: [package]
+tags: [ellmer]
+---
+
+<!--
+TODO:
+* [x] Look over / edit the post's title in the yaml
+* [x] Edit (or delete) the description; note this appears in the Twitter card
+* [x] Pick category and tags (see existing with `hugodown::tidy_show_meta()`)
+* [x] Find photo & update yaml metadata
+* [x] Create `thumbnail-sq.jpg`; height and width should be equal
+* [x] Create `thumbnail-wd.jpg`; width should be >5x height
+* [x] `hugodown::use_tidy_thumbnails()`
+* [x] Add intro sentence, e.g. the standard tagline for the package
+* [ ] `usethis::use_tidy_thanks()`
+-->
+
+We're very happy to announce the release of [ellmer](https://ellmer.tidyverse.org) 0.4.0. ellmer makes it easy to chat with a large language model directly from R. ellmer is an R package designed to make it easy to use large language models (LLMs) from R. It supports a wide variety of providers (including OpenAI, Anthropic, Azure, Google, Snowflake, Databricks and many more), makes it easy to [extract structured data](https://ellmer.tidyverse.org/articles/structured-data.html), and to give the LLM the ability to call R functions via [tool calling](https://ellmer.tidyverse.org/articles/tool-calling.html).
+
+You can install it from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("ellmer")
+```
+
+This blog post will cover the major changes in this release, including important lifecycle updates, new features for Claude (caching, file uploads, and web tools), improvements to OpenAI support (responses API and built-in tools), and a variety of enhancements to error handling, pricing tracking, and security.
+
+You can see a full list of changes in the [release notes](https://github.com/tidyverse/ellmer/releases/tag/v0.4.0).
+
+```{r setup}
+library(ellmer)
+```
+
+## Lifecycle
+
+* `parallel_chat()` and `batch_chat()` are no longer experimental. Based on user feedback, both `parallel_chat()` and `batch_chat()` do a much better job of handling errors, and I'm confident that they're around to stay.
```

Comment:
Maybe this and `New features` could do with one less level of bullet points, since each is self-contained in a paragraph already?